### DISCUSSION TITLE
MicroOS package manager?

### DISCUSSION THREAD

**Discussion by TimHeckel** [mysticaltech/terraform-hcloud-kube-hetzner#274] [Q&A] [answered]
Hey all - this is a basic question, but I'm struggling to install `vi` or any external package really using a package manager when ssh'd into a MicroOS container. I've found [this page ](https://en.opensuse.org/Portal:MicroOS/Zypper)regarding MicroOS and Zypper, and it seems like `zypper` would be the expected package manager, but alas there is no `zypper` in the command line, and I haven't yet figured out how to get `vi` installed...any help?

**Comment by mysticaltech** [comment, marked as answer]
@TimHeckel Don't worry about the underlying workings of zypper, microOS is a transactional system. Just use the `extra_packages_to_install=['vi']` flag.

**Reply by TimHeckel** [reply]
Thank you as always @mysticaltech  :) So....is `extra_packages_to_install` an option in the `kube.tf` or elsewhere? Ideally I'd love to have some packages installed by default whenever I ssh directly into the containers. I did find that `zypper` is available when I ssh into the machine directly that Hetzner provides, but I'd like `vi` or other packages to be available when I ssh into the containers themselves.

**Reply by mysticaltech** [reply]
Yes, the flag is for the nodes, in kube.tf of course (no other files are necessary to deploy the cluster).

Now, vi at the container level is not something related to this project. You have to mod the containers you are using themselves.
